src/main/java/appeng/tile/inventory/AppEngInternalAEInventory.java

```diff
@@ -214,50 +248,20 @@ else if( oldStack.getCount() < newItemStack.getCount() )
 					removed = added = ItemStack.EMPTY;
 				}
 			}
-
-			this.te.onChangeInventory( this, slot, InvOperation.setInventorySlotContents, removed, added );
+			if( this.te != null && Platform.isServer() )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2589)